hphp/runtime/vm/jit/abi-arm.h

```diff
@@ -144,6 +165,46 @@ inline jit::ConditionCode convertCC(vixl::Condition cc) {
   return mapping[cc];
 }
 
+enum class StatusFlags : uint8_t {
+  None = 0,
+  N    = 1 << 3,
+  C    = 1 << 2,
+  Z    = 1 << 1,
+  V    = 1,
+  CZ   = C | Z,
+  NV = N | V,
+  NZV = N | Z | V,
+  NotC = N | Z | V,
+  NotV = N | C | Z,
+  All  = N | C | Z | V,
+};
+
+inline uint8_t required_flags(jit::ConditionCode cc) {
+  assertx(cc >= 0 && cc <= 0xF);
+
+  // A jit cc is first converted to an ARM cc before using this mapping
+  constexpr StatusFlags mapping[] = {
+    StatusFlags::Z,         // eq
+    StatusFlags::Z,         // ne
+    StatusFlags::C,         // hs
+    StatusFlags::C,         // lo
+    StatusFlags::N,         // mi
+    StatusFlags::N,         // pl
+    StatusFlags::V,         // vs
+    StatusFlags::V,         // vc
+    StatusFlags::CZ,        // hi
+    StatusFlags::CZ,        // ls
+    StatusFlags::NV,        // ge
+    StatusFlags::NV,        // lt
+    StatusFlags::NZV,       // gt
+    StatusFlags::NZV,       // le
+    StatusFlags::None,      // nv (invalid)
+    StatusFlags::None,      // al (invalid)
+  };
```

Comment:
This is ARM's implementation of the platform-specific function which maps a vasm CC to the exact flags which are required to be set for the Vinstr def in order for the Vinstr use to test that condition code.